### ISSUE TITLE
Issue with vertical rule being offset

### ISSUE THREAD

**Issue by astrofrog** [aperiosoftware/timeseries.js#33]
In the following JSON file:

https://gist.github.com/astrofrog/8ca5a1a324f47db02f53d4c8ca5a8a1a

the vertical rule should appear on:

```
            "signal": "datetime(2009, 5, 2, 0, 51, 28)"
```

(i.e. May 2nd 2009) but it appears to be offset:

<img width="764" alt="screenshot 2019-01-18 at 16 43 37" src="https://user-images.githubusercontent.com/314716/51400406-399da380-1b40-11e9-8979-c8bd7bcdd2e4.png">

**Comment by slowe** [comment]
That looks as though it is because my implementation tries to follow [the VEGA docs which uses a zero-indexed month](https://vega.github.io/vega/docs/expressions/#datetime).

> datetime(year, month[, day, hour, min, sec, millisec])
Returns a new Date instance. The month is 0-based, such that 1 represents February.

So, for May 2nd, you'd have to use `"signal": "datetime(2009, 4, 2, 0, 51, 28)"`;

In checking this I found a small bug when one of the datetime values was `0`. In the example above the time part was not being included because the hours were set to zero. That is now fixed.
